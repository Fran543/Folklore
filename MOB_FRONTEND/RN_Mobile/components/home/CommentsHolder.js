import { View, StyleSheet, FlatList } from "react-native";
import { useState } from 'react'
import Comment from "./Comment";

export default function CommentsHolder() {
    const [comments, setComments] = useState([{ IDComment: 1, Username: "Ivo Ivic", Content: "komentarkomentarkomentar" },
    { IDComment: 2, Username: "Sara Ivic", Content: "komentarkomentarkomentar" },
    { IDComment: 3, Username: "Marica Ivic", Content: "komentarkomentarkomentar" }])

    function renderCommentItem(itemData) {
        const commentItemProps = {
            IDComment: itemData.item.IDComment,
            Username: itemData.item.Username,
            Content: itemData.item.Content
        }

        return <Comment {...commentItemProps} />
    }

    return (
        <View style={styles.content}>
            <FlatList data={comments} renderItem={renderCommentItem}
                keyExtractor={(item, index) => {
                    return item.IDComment
                }}
            />
        </View>
    );
}

const styles = StyleSheet.create({
    content: {
        flex: 1,
        paddingTop: 30
    },
});