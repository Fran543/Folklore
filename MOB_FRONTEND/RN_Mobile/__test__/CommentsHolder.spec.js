import React from 'react';
import renderer from 'react-test-renderer';

import CommentsHolder from '../components/home/CommentsHolder';

describe('<CommentsHolder />', () => {

    it('renders correctly', async () => {
        const tree = await renderer.create(<CommentsHolder idStory="1" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('has 1 child', async () => {
        const tree = await renderer.create(<CommentsHolder idStory="1"/>).toJSON();
        expect(tree.children.length).toBe(1);
    });
});
