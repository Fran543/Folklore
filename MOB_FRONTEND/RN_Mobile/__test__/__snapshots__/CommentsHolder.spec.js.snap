// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentsHolder /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 30,
    }
  }
>
  <RCTScrollView
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</View>
`;
